recipes/optick/all/conanfile.py

```diff
@@ -0,0 +1,64 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.microsoft import is_msvc_static_runtime, is_msvc
+from conan.tools.files import get, copy, rm, rmdir
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
+import os
+
+required_conan_version = ">=1.54.0"
+
+class PackageConan(ConanFile):
+    name = "optick"
+    description = "Optick is a super-lightweight C++ profiler for Games"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/bombomby/optick"
+    topics = ("profiler", "games")
+    package_type = "shared-library"
+    settings = "os", "arch", "compiler", "build_type"
+
+    @property
+    def _min_cppstd(self):
+        return 11
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        pass
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        if not is_msvc(self):
+            raise ConanInvalidConfiguration(f"{self.ref} can be built only by Visual Studio and msvc.")
```

Comment:
@luizfeldmann Why is it not working for other platforms. Looking at https://github.com/bombomby/optick/blob/1.4.0.0/CMakeLists.txt I see no restrictions.

Reply:
> The profiler library, which is currently built in this recipe

Yes, and as far as I see, it generates for any platform:

- Windows:
```
optick/1.4.0.0: RUN: cmake --install "C:\J2\w\prod-v2\bsr@3\36200\eeccb\p\b\optic78d0377885876\b\build" --config Release --prefix "C:/J2/w/prod-v2/bsr@3/36200/eeccb/p/b/optic78d0377885876/p"
-- Installing: C:/J2/w/prod-v2/bsr@3/36200/eeccb/p/b/optic78d0377885876/p/lib/OptickCore.lib
-- Installing: C:/J2/w/prod-v2/bsr@3/36200/eeccb/p/b/optic78d0377885876/p/bin/OptickCore.dll
-- Installing: C:/J2/w/prod-v2/bsr@3/36200/eeccb/p/b/optic78d0377885876/p/include/optick.h
-- Installing: C:/J2/w/prod-v2/bsr@3/36200/eeccb/p/b/optic78d0377885876/p/include/optick.config.h
-- Installing: C:/J2/w/prod-v2/bsr@3/36200/eeccb/p/b/optic78d0377885876/p/lib/cmake/Optick/OptickTargets.cmake
-- Installing: C:/J2/w/prod-v2/bsr@3/36200/eeccb/p/b/optic78d0377885876/p/lib/cmake/Optick/OptickTargets-release.cmake
-- Installing: C:/J2/w/prod-v2/bsr@3/36200/eeccb/p/b/optic78d0377885876/p/lib/cmake/Optick/OptickConfig.cmake
-- Installing: C:/J2/w/prod-v2/bsr@3/36200/eeccb/p/b/optic78d0377885876/p/lib/cmake/Optick/OptickConfigVersion.cmake
```

- Mac
```
Install the project...
cmake -P cmake_install.cmake
-- Install configuration: "Release"
-- Installing: /tmp/install/lib/libOptickCore.dylib
-- Installing: /tmp/install/include/optick.h
-- Installing: /tmp/install/include/optick.config.h
-- Installing: /tmp/install/lib/cmake/Optick/OptickTargets.cmake
-- Installing: /tmp/install/lib/cmake/Optick/OptickTargets-release.cmake
-- Installing: /tmp/install/lib/cmake/Optick/OptickConfig.cmake
-- Installing: /tmp/install/lib/cmake/Optick/OptickConfigVersion.cmake
```

So it should not block other platforms, as the main objective is package libraries and headers.

> The GUI to receive the profiler data, which is a C# project and must build on Windows

How does it work? Is it a define in the header, a configuration file? Who generates it?